Modules/CodeTrek/Http/Controllers/CodeTrekController.php

```diff
@@ -49,16 +50,22 @@ public function show($id)
     /**
      * Show the form for editing the specified resource.
      */
-    public function edit($id)
+    public function edit(CodeTrekService $service, $id)
```

Comment:
Laravel automatically provides you with the object of the model if you define the input type in the controller function. This concept is known as model binding in laravel.

```suggestion
    public function edit(CodeTrekService $service, CodeTrekApplicant $applicant)
```

Reply:
Just take care that the name of the input argument and the name of dynamic variable define in route should be same